### DISCUSSION TITLE
Navidrome Music Mount not available on boot, removing bookmark and playlist items

### DISCUSSION THREAD

**Discussion by SickProdigy** [navidrome/navidrome#3310] [General]
Hello, I've run into an issue sort of and figure I'd come here,
first: discord links aren't working on website or github.

Actual problem:
I was upgrading some drives on my pc, didn't have my music folder mounted, and I guess navidrome did a scan.
After the scan was complete, all of my ratings and playlists were emptied. I've run into a similar issue with just moving the library to a different folder. Navidrome scans and empties all of my data.

I was hoping there might be multiple copies within the table but looking at the table, it's just fresh data. No bookmarks.

I've been racking my brain all day trying to figure out if it's recoverable, but at this point I don't think so :(

Well I'm here now to hopefully prevent this from happening again. Does anyone have any suggestions? i'm definitely setting up some backups, maybe even a git to actively monitor the database so I can see when it changed and what changed.

**Comment by deluan** [comment]
Sorry, but if you don't have a backup, there's nothing much that can be done :/

Navidrome has a safeguard for when your music folder is _completely_ empty, usually the result of it not being mounted. When it detects a _completely_ empty music folder, it cancel the scan process. But if your unmounted music folder has _any_ file in it, it won't be considered empty and the scan will happen, removing from the DB any music file that is not present at time of scan.

In the new scanner I'm working on (#2709), the songs will be marked as missing, but won't be deleted from the DB.

**Reply by SickProdigy** [reply]
Ah yes I see now. I had another software adding things to the wrong drive which in turn showed new items in my music folder.

Alright, I appreciate your quick response and forwardness. Glad to hear of a preventive measure in the future!

I think I will try to add some command line options to my docker file to take backups every boot or something similar. Open to suggestions!

**Reply by deluan** [reply]
This is a good plan. And we will be adding automatic backups (#3194) in a future release.
